lib/fluent/plugin_helper/http_server/server.rb

```diff
@@ -38,7 +38,8 @@ def initialize(addr:, port:, logger:, default_app: nil, tls_context: nil)
           scheme = tls_context ? 'https' : 'http'
           @uri = URI("#{scheme}://#{@addr}:#{@port}").to_s
           @router = Router.new(default_app)
-          @reactor = Async::Reactor.new(nil, logger: @logger)
+          @server_task = nil
+          Console.logger = @logger
```

Comment:
Console.logger doesn't seem compatible with Ruby's Logger, we need to address it.
It's also affected current master, sometimes following error occurs especially on macOS:
```
2022-07-01T01:29:37.2657850Z 	from /Users/runner/hostedtoolcache/Ruby/2.7.6/x64/lib/ruby/gems/2.7.0/gems/async-1.30.3/lib/async/task.rb:242:in `fail!'
2022-07-01T01:29:37.2658430Z Failure: test: with insecure in transport section(HttpHelperTest::Create a HTTP server::create a HTTPS server::#http_server_create_https_server)
2022-07-01T01:29:37.2694380Z 	from /Users/runner/hostedtoolcache/Ruby/2.7.6/x64/lib/ruby/gems/2.7.0/gems/async-1.30.3/lib/async/task.rb:267:in `rescue in block in make_fiber'
2022-07-01T01:29:37.2694980Z /Users/runner/work/fluentd/fluentd/test/plugin_helper/test_http_server_helper.rb:296:in `block (5 levels) in <class:HttpHelperTest>'
2022-07-01T01:29:37.2786170Z 	from /Users/runner/hostedtoolcache/Ruby/2.7.6/x64/lib/ruby/gems/2.7.0/gems/async-1.30.3/lib/async/task.rb:264:in `block in make_fiber'
2022-07-01T01:29:37.2786710Z      293:             assert_equal('200', resp.code)
2022-07-01T01:29:37.2820540Z /Users/runner/hostedtoolcache/Ruby/2.7.6/x64/lib/ruby/gems/2.7.0/gems/async-io-1.33.0/lib/async/io/generic.rb:216:in `accept_nonblock': SSL_accept returned=1 errno=0 state=error: tlsv1 alert unknown ca (OpenSSL::SSL::SSLError)
2022-07-01T01:29:37.2821140Z      294:             assert_equal('hello get', resp.body)
2022-07-01T01:29:37.2823720Z 	from /Users/runner/hostedtoolcache/Ruby/2.7.6/x64/lib/ruby/gems/2.7.0/gems/async-io-1.33.0/lib/async/io/generic.rb:216:in `async_send'
2022-07-01T01:29:37.2824260Z      295: 
2022-07-01T01:29:37.2826690Z 	from /Users/runner/hostedtoolcache/Ruby/2.7.6/x64/lib/ruby/gems/2.7.0/gems/async-io-1.33.0/lib/async/io/generic.rb:62:in `block in wrap_blocking_method'
2022-07-01T01:29:37.2827090Z   => 296:             assert_raise OpenSSL::SSL::SSLError do
2022-07-01T01:29:37.2828630Z 	from /Users/runner/hostedtoolcache/Ruby/2.7.6/x64/lib/ruby/gems/2.7.0/gems/async-io-1.33.0/lib/async/io/ssl_socket.rb:143:in `block in accept'
2022-07-01T01:29:37.2830080Z      297:               secure_get("https://127.0.0.1:#{@port}/example/hello")
2022-07-01T01:29:37.2903090Z 	from /Users/runner/hostedtoolcache/Ruby/2.7.6/x64/lib/ruby/gems/2.7.0/gems/async-1.30.3/lib/async/task.rb:261:in `block in make_fiber'
2022-07-01T01:29:37.2903440Z      298:             end
2022-07-01T01:29:37.2974280Z      299:           end
2022-07-01T01:29:37.2980450Z /Users/runner/work/fluentd/fluentd/test/plugin_helper/test_http_server_helper.rb:36:in `on_driver'
2022-07-01T01:29:37.2983740Z /Users/runner/work/fluentd/fluentd/test/plugin_helper/test_http_server_helper.rb:66:in `on_driver_transport'
2022-07-01T01:29:37.2984210Z /Users/runner/work/fluentd/fluentd/test/plugin_helper/test_http_server_helper.rb:286:in `block (4 levels) in <class:HttpHelperTest>'
2022-07-01T01:29:37.2984460Z 
2022-07-01T01:29:37.2987480Z <OpenSSL::SSL::SSLError> expected but was
2022-07-01T01:29:37.2989170Z <ArgumentError(<wrong number of arguments (given 3, expected 0..1)>)
2022-07-01T01:29:37.2989610Z /Users/runner/hostedtoolcache/Ruby/2.7.6/x64/lib/ruby/2.7.0/logger.rb:536:in `warn'
2022-07-01T01:29:37.2990540Z /Users/runner/hostedtoolcache/Ruby/2.7.6/x64/lib/ruby/gems/2.7.0/gems/async-1.30.3/lib/async/task.rb:242:in `fail!'
2022-07-01T01:29:37.2991210Z /Users/runner/hostedtoolcache/Ruby/2.7.6/x64/lib/ruby/gems/2.7.0/gems/async-1.30.3/lib/async/task.rb:267:in `rescue in block in make_fiber'
2022-07-01T01:29:37.2992000Z /Users/runner/hostedtoolcache/Ruby/2.7.6/x64/lib/ruby/gems/2.7.0/gems/async-1.30.3/lib/async/task.rb:264:in `block in make_fiber'
2022-07-01T01:29:37.2992290Z >
2022-07-01T01:29:37.2992390Z 
2022-07-01T01:29:37.2992450Z diff:
2022-07-01T01:29:37.2992690Z ? Op   enSSL::SSL::SSLError                                                      
2022-07-01T01:29:37.2992980Z ? Argum  t                 (<wrong number of arguments (given 3, expected 0..1)>)
2022-07-01T01:29:37.2993260Z ? ??  ?????????????                 ++++++++++++++++++++++++++++++++++++++++++++++++++++++
2022-07-01T01:29:37.2993530Z + /Users/runner/hostedtoolcache/Ruby/2.7.6/x64/lib/ruby/2.7.0/logger.rb:536:in `warn'
2022-07-01T01:29:37.2994060Z + /Users/runner/hostedtoolcache/Ruby/2.7.6/x64/lib/ruby/gems/2.7.0/gems/async-1.30.3/lib/async/task.rb:242:in `fail!'
2022-07-01T01:29:37.2994730Z + /Users/runner/hostedtoolcache/Ruby/2.7.6/x64/lib/ruby/gems/2.7.0/gems/async-1.30.3/lib/async/task.rb:267:in `rescue in block in make_fiber'
2022-07-01T01:29:37.2995800Z + /Users/runner/hostedtoolcache/Ruby/2.7.6/x64/lib/ruby/gems/2.7.0/gems/async-1.30.3/lib/async/task.rb:264:in `block in make_fiber'
```

Reply:
I'm working on it in #3987

Reply:
merged.